### ISSUE TITLE
Konzeptaufgabe: Swing

### ISSUE THREAD

**Issue by AMatutat** [PM-Dungeon/PM-Lecture#122]

**Comment by AMatutat** [comment]
Dungeon Map Creator.

**Comment by AMatutat** [comment]
@cagix Swing fällt auf die letzte "richtige" Vorlesungswoche. Daher gibt es kein neues Praktikum für Swing.
Ich finde die Aufgabe aber nett. Man könnte die in eine "richtige" Dungeon Aufgabe verwandeln und mit dem Vermerk "Selbststudium" freigeben. Entsprechend bepunktet natürlich.

**Comment by cagix** [comment]
ja, oder swing nochmal nach vorn schieben?

**Comment by cagix** [comment]
nochmal etwas weiter gedacht: die level sind ja in json, oder? d.h. man könnte das mit einer lib/api für json verbinden. die sollen darüber ein level laden, umbauen und speichern.

**Comment by AMatutat** [comment]
> nochmal etwas weiter gedacht: die level sind ja in json, oder? d.h. man könnte das mit einer lib/api für json verbinden. die sollen darüber ein level laden, umbauen und speichern.

nein, sind sie (noch) nicht. Hab mich nicht weiter darum gekümmert.... Jonas hat aber auch den Eindruck erweckt, dass er das brauchen könnte... Vielleicht sollte ich das fertig machen. https://github.com/PM-Dungeon/core/issues/166

**Comment by cagix** [comment]
hmmm. im prinzip ja. aber eigentlich sprechen wir ja immer noch über ne konzeptaufgabe. hmmm. insofern könnte man hinter swing/json auch einen anderen use-case klemmen?

**Comment by cagix** [comment]
oder halt statt json ne andere lib nehmen ...

**Comment by AMatutat** [comment]
Ich glaube wir verlieren grade topic.
Die Frage ist ja: Wollen wir eine Swing Übung machen? 
Wenn ja, müssen wir Swing nach vorne schieben, wenn nicht, dann machen wir hier dicht und gut ist.

**Comment by cagix** [comment]
eine übung wäre nett, und es wäre auch eine derjenigen, die sich leichter korrigieren lassen (quasi durch anschauen/ausprobieren).

aber wenn ich auf den plan schaue, sehe ich nicht wirklich viel spielraum. man könnte frameworks, swing, java2d (woche 12) gegen ant, maven, docker (woche 8) tauschen. diese drei sachen sind ja eigentlich nur zur vollständigkeit drin, da wird es keine übungen geben und auch keine prüfungsfragen. d.h. man könnte ant, maven, docker nach woche 12 schieben und dafür in swing und irgendeiner lib eine übung machen. 

was denkst du? der kram in der letzten woche ist natürlich massiv gefährdet: das ist das zeugs, was wegfällt, wenn vorher irgendwas umgeplant werden muss.

**Comment by AMatutat** [comment]
Woche 8 mit 12 tauschen klingt gut. Dann wandert die Swing Aufgabe nach Sheet07.
